src/core/lib/iomgr/timer_generic.cc

```diff
@@ -404,15 +420,22 @@ static void timer_init(grpc_timer* timer, grpc_millis deadline,
   if (is_first_timer) {
     gpr_mu_lock(&g_shared_mutables.mu);
     if (grpc_timer_trace.enabled()) {
-      gpr_log(GPR_DEBUG, "  .. old shard min_deadline=%" PRIdPTR,
+      gpr_log(GPR_DEBUG, "  .. old shard min_deadline=%" PRId64,
               shard->min_deadline);
     }
     if (deadline < shard->min_deadline) {
-      gpr_atm old_min_deadline = g_shard_queue[0]->min_deadline;
+      grpc_millis old_min_deadline = g_shard_queue[0]->min_deadline;
       shard->min_deadline = deadline;
       note_deadline_change(shard);
       if (shard->shard_queue_index == 0 && deadline < old_min_deadline) {
+#if GPR_ARCH_64
         gpr_atm_no_barrier_store(&g_shared_mutables.min_timer, deadline);
```

Comment:
Now that deadline and other uses are grpc_millis, shouldn't there be a static_cast or something to gpr_atm in these cases? Since the arguments to atm_store/load are supposed to be gpr_atm afaik.